src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1306,7 +1306,25 @@ def from_strings(cls, strings):
         >>> PlacementSpec.from_strings(None)
         PlacementSpec()
         """
-        strings = strings or []
+        if arg is None:
+            strings = []
+        elif isinstance(arg, str):
+            if ' ' in arg:
+                strings = arg.split(' ')
+            elif ';' in arg:
+                strings = arg.split(';')
+            elif ',' in arg and '[' not in arg:
+                # FIXME: this isn't quite right.  we want to avoid breaking
+                # a list of mons with addrvecs... so we're basically allowing
+                # , most of the time, except when addrvecs are used.  maybe
+                # ok?
+                strings = arg.split(',')
+            else:
+                strings = [arg]
+        elif isinstance(arg, list):
```

Comment:
weird, this conflicts with the annotation of this method signature where `arg` can only be `None` or `str`.

Reply:
yeah, i'm waiting on #33476 and also coding defensively... but probably best to let the other PR merge and then adjust all callers to pass a string.

Reply:
rebased and cleaned up